PlatformRequirements.md

```diff
@@ -0,0 +1,144 @@
+        - buildOptions.release : This indicates that packages should be built with release configuration. If not set to true, debug configuration should be used.
+        - buildOptions.device : This indicates that the built app is intended to run on device.
+        - buildOptions.emulator : This indicates that the built app is intended to run on an emulator.
+        - buildOptions.target : This indicates the device id that will be used to run the built app.
+        - buildOptions.nobuild : This indicates that this should be a dry-run call. No build artifacts should be produced.
+        - buildOptions.archs : This indicates chip architectures with app packages should be built for. The list of valid architectures is dependent on the platform.
+        - buildOptions.buildConfig : This is the path to the build configuration file. The format of this file is dependent on the platform.
+        - buildOptions.argv : This is a raw array of command-line arguments that should be passed to the `build` command. The purpose of this property is to pass platform-specific arguments, and eventually let the platform define its own arguments processing logic.
+        - The `build` method must return a promise either fulfilled with an array of build artifacts (application packages) if the package was built successfully, or rejected with a CordovaError. The return value in most cases will contain only one items, but in some cases there could be multiple items in an output array, e.g. when multiple architectures are specified. The resultant build artifact objects are not strictly typed and may contain an arbitrary set of fields as in the sample below.
+ 
+        ```
+        {
+            architecture: 'x86',
+            buildType: 'debug',
+            path: '/path/to/build',
+            type: 'app'
+ 
+        }
+        ```
+ 
+- has a `run` method
+    - PlatformApi.prototype.run = function(runOptions) {};
+    - The `run` method should build an application package for the current platform and runs it on the specified/default device. If no 'device'/'emulator'/'target' options are specified, then it should try to run the app on a default device if connected, otherwise it should run on the app on the emulator.
+    - runOptions : This is an options object. The structure is the same as for build options.
+    - The `run` method must return a promise either fulfilled if the package was build and ran successfully, or rejected with a CordovaError.
+ 
+ 
+- has an `addPlugin` method
+    - PlatformApi.prototype.addPlugin = function (plugin, installOptions) {};
+    - The `addPlugin` method should install a plugin into a platform. It should handle all the non-www files shipped by plugin (sources, libs, assets, js-files) and accept a PluginInfo instance that represents the plugin that will be installed and an options object. It cannot resolve the dependencies of a plugin.
```

Comment:
Yeah totally! The Platform API would be consumed by cordova tooling such as cordova-cli and lib, so linking directly to code I think is appropriate. The folks writing up Platform API implementations would reasonably need to know what properties and/or methods exist on these objects being passed in (like `PluginInfo`) to the Platform API methods they are implementing.